src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java

```diff
@@ -144,4 +155,60 @@ public FileDownloadController(ExperimentFileLocationService experimentFileLocati
             zipOutputStream.close();
         }
     }
+
+    @GetMapping(value = "json/experiments/download/zip/check",
+            produces = "application/json;charset=UTF-8")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public String
+    checkMultipleExperimentsFileValid(@RequestParam(value = "accession", defaultValue = "") List<String> accessions)
+    {
+        List<Experiment> experiments = new ArrayList<>();
+
+        accessions.stream().forEach(accession -> {
+            try{
+                experiments.add(experimentTrader.getPublicExperiment(accession));
+            } catch (Exception e) {
+                LOGGER.debug("Invalid experiment accession: {}", accession);
+            }
+        });
```

Comment:
I’m not really sure that we want to guard ourselves against bad accessions. Since this URL is going to be built and checked from the experiments table component, and is a helper endpoint, we have a reasonable guarantee that it will only contain good accessions, and returning a 404 if a bad accession is provided might not be that bad.

In any case, you be the judge. But whatever choice you make, a couple of points:
- Use `var`.
- Add a blank between try and opening curly brace.
- No need to use `stream()` with `forEach()`.
- Use an `ImmutableList.Builder` or, better yet, since you don’t want duplicates and you’re not accessing elements by index, just go for an `ImmutableSet.Builder`.